interview-questions.md

```diff
@@ -8,36 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: First of all, your <Student> table will have the foreign key because your <Cohort> table has_many students. Therefore, that tells me that your <Student> table will need a foreign key to Match the ID of the <Cohort> it belongs to. If you were creating a project and this was your DB structore and you forgot your foreign key column in your <Student> table (FEAR NOT RUBY ON RAILS HAS YOUR BACK); you will need to generate a new migration by running $rails g migration add_foreign_key_to_student cohort_id:integer. This will create a migration file and you can add in the file: <add_column :student, cohort_id, integer>and then you will need to run $rails db:migrate this will document and make the changes happen to your <Student> table. Dont ever physically go into your schema and change your table.
 
 Researched answer:
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: GET(when using an show method), POST, PATCH/PUT and DELETE.
 
-Researched answer:
+Researched answer: POST requests to the create action: When creating a new resource, you must pass parameters to set the initial values for the resource's attributes. PATCH/PUT requests to the update action: When updating an existing resource, you must pass parameters to indicate which attributes to update and their new values. DELETE requests to the destroy action: When deleting a resource, you must pass a parameter to identify which resource to delete.
```

Comment:
I like the way you describe delete and mention that we need to identify which data to delete.